src/impl_/trampoline.rs

```diff
@@ -220,7 +220,7 @@ where
     let py_err = match panic_result {
         Ok(Ok(value)) => return value,
         Ok(Err(py_err)) => py_err,
-        Err(payload) => PanicException::from_panic_payload(payload),
+        Err(payload) => PanicException::from_panic_payload(py, payload),
```

Comment:
The types do not work out at all for now, but I wanted to discuss the concrete approach first before trying plough through the trampolines.